ui/src/StateContainer.js

```diff
@@ -30,9 +31,10 @@ export class StateContainer extends LitElement {
         return {
             view: {type: String},
             canRecord: {type: Boolean},
-            isRecording: {type: Boolean},
             audioStream: {type: Object},
             toast: {type: Object},
+            collectionState: {type: String},
+            context: {type: Object},
```

Comment:
+1 to single context.
You can probably initialize the context in this class and pass it into the record button component. It looks like the record button already receives context as a property.

It shouldn't have to make a new one every time on line 82 recordbutton.js